### PR TITLE
Parse annotations

### PR DESCRIPTION
**Pull request by tcmitchell**
Parse annotation objects on the path to extensions (see #7).

### CODE REVIEW

File: sbol2/document.py [hunk 1]
```diff
@@ -545,11 +554,7 @@ def parse_objects_inner(self, subject, obj):
             new_obj.doc = self
 
     def parse_properties_inner(self, subject, predicate, obj):
-        if self.logger.isEnabledFor(logging.DEBUG):
-            self.logger.debug("Adding: (" + str(subject) + " - " +
```
Comment (bbartley) [inline sbol2/document.py:549]: This makes my eyes hurt.  Glad you changed it.

File: sbol2/document.py [hunk 2]
```diff
@@ -545,11 +554,7 @@ def parse_objects_inner(self, subject, obj):
...
-                              str(type(subject)) + ", " + str(predicate) +
-                              " - " + str(type(predicate)) + ", " + str(obj) +
-                              " - " + str(type(obj)) + ")")
```
Comment (bbartley) [inline sbol2/document.py:552]: This makes my eyes hurt.  Glad you changed it.

File: sbol2/document.py [hunk 3]
```diff
@@ -581,13 +586,72 @@ def parse_properties_inner(self, subject, predicate, obj):
...
+        for ao in annotation_objects:
+            self.logger.debug('Annotation object: %s', ao.identity)
+            if SBOL_PERSISTENT_IDENTITY in ao.properties:
+                # Copy to a new TopLevel object
+                tl = TopLevel(type_uri=ao.type)
+                for k, v in ao.properties.items():
+                    tl.properties[k] = v
+                for k, v in ao.owned_objects.items():
+                    tl.owned_objects[k] = v
+                tl.doc = self
```
Comment (bbartley) [inline sbol2/document.py:617]: This is a case where the `cast` method would be helpful, e.g.:
```
tl = ao.cast(TopLevel)
```
No matter, this code block accomplishes the same thing.

Comment (tcmitchell) [reply sbol2/document.py:617]: One of us is confused. I don't think the `cast` operator creates a new object. The C++ code [creates a new object in this case](https://github.com/SynBioDex/libSBOL/blob/master/source/document.cpp#L895-L903), and this python code is a direct translation of that. The C++ code does not use the cast operator here.

Comment (bbartley) [reply sbol2/document.py:617]: The point I'm trying to make is that this pattern tends to show up in different places.  This code creates a new object of type TopLevel, then copies all the properties from the base SBOLObject, and then replaces that SBOLObject (bye, bye!) with the new TopLevel.  Effectively, that means you have downcast from an object of type SBOLObject to TopLevel.  That's what the `cast` methods were implemented for:
https://github.com/SynBioDex/libSBOL/blob/develop/source/object.cpp#L78
Just saying, it's a pattern that shows up now and again, where the `cast` method can be used to accomplish the same thing in a more explicit manner.  This is more of an FYI than a request for a code change.

Comment (tcmitchell) [reply sbol2/document.py:617]: ❗  I did not know that. I thought cast was doing something with the underlying libSBOL object. Namely, just casting it to a different type. I didn't know it was actually creating a new object. It's more like a copy than a cast. Maybe.

**Pull request merged**